mindsdb/integrations/handlers/firestore_handler/firestore_handler.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any, Dict, Union
+from mindsdb.integrations.libs.base import DatabaseHandler
+from mindsdb.integrations.libs.response import HandlerStatusResponse
+
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import firestore
+
+
+class FirestoreHandler(DatabaseHandler):
+    name = "firestore"
+
+    def __init__(
+        self,
+        name: str,
+        cert: Union[str, None] = None,
+        options: Dict[str, Any] = {},
+    ):
+        """
+        __init__ Initialize the FirestoreHandler class
+
+        Parameters
+        ----------
+        name : str
+            The name of the firestore app.
+        cert : str
+            The path to the certificate file or a dict representing the contents of the certificate.
+            Defaults to the Google Application Default Credentials
+        options : Dict[str, Any], optional
+            _description_, by default {}
+        """
+        super().__init__(name)
+
+        self.name = name
+        self.app = None
+        self.db = None
+        if cert is None:
+            self.__cred = credentials.ApplicationDefault()
```

Comment:
This will not work. Please check other handlers for how to get the values from the CREATE DATABASE statement